context/EggheadUserProvider.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react'
```

Comment:
all of this is imported and frankly it feels confusing and could use some attention and thought 🤔

Reply:
I know the full import for "React" is required when the babel config looks for the jsx pragma.

I'm fairly certain Next.js uses:

https://github.com/vslinko/babel-plugin-react-require

to automatically inject "React" to satisfy the babel/jsx/pragma lookup.